### ISSUE TITLE
Why does "wrap" require a parameter when a "func" may not need one?

### ISSUE THREAD

**Issue by deanebarker** [lunet-io/scriban#177]
Say I have an HTML construct that I put in function:

```
{{ func aside }}
  <aside class="whatever">
    {{ $$ }}
  </aside>
{{ end }}
```

I want to call it like this:

```
{{ wrap aside }}
  This is some content for the aside.
{{ end }}
```

However, this throws a parsing error because there is no extra parameter on the `wrap` call, even though I don't need one:

```
Expecting a direct function instead of the expression 'aside/variable'
```

Why is this?  I don't need any extra parameters except what's in the `wrap` block itself. What I end up doing is calling it like this:

```
{{ wrap aside '' }}
```

Essentially, I pass an extra parameter as an empty string, and just don't do anything with it. But, clearly, this is hack-ish and not something I want to get in the habit of doing.

What am I not understanding here?

**Comment by xoofx** [comment]
Maybe that's just a bug

**Comment by deanebarker** [comment]
I'll leave that to you to judge, but if so, it's a fairly benign one, because there's a workaround.

But, good to know.

**Comment by deanebarker** [comment]
Here's the line it's coming from:

https://github.com/lunet-io/scriban/blob/master/src/Scriban/Syntax/ScriptWrapStatement.cs#L26

**Comment by xoofx** [comment]
Right, it's not a big but the documentation is actually wrong as you need to add a `@` before the function: `wrap @caption`

**Comment by deanebarker** [comment]
I assumed that the _only_ thing that could go there is a function name, so what is the `@` meant to represent?  That makes it seem that the function name is meant to be differentiated from something else -- what would that something else be?

**Comment by deanebarker** [comment]
Also, this didn't seem to work. This code...

```
{{ wrap @aside }}
```
...now throws this error:
```
Expecting a direct function instead of the expression `FunctionAliasaside/unary expression`
```

**Comment by xoofx** [comment]
> I assumed that the only thing that could go there is a function name, so what is the @ meant to represent? That makes it seem that the function name is meant to be differentiated from something else -- what would that something else be?

Not exactly, you could have a function that returns a function pointer.  Maybe things got broken after I introduced function pointers.

> Also, this didn't seem to work. This code...

Yeah, sorry, I don't know what's going on. I'm on a business trip for the next 2 weeks and I can't check more deeply that issue for now, unless you go into it before.

**Comment by deanebarker** [comment]
No worries. I have a workaround.

I did try to look into the code, but I got lost fairly quickly. Just know that I tried.

**Comment by xoofx** [comment]
Actually got a chance to have a look at it and it should be fixed by commit 287d936

**Comment by xoofx** [comment]
The fix will be available in an upcoming release
